pkg/datapath/linux/config/config.go

```diff
@@ -82,6 +84,7 @@ func NewHeaderfileWriter(p WriterParams) (datapath.ConfigWriter, error) {
 		log:                p.Log,
 		sysctl:             p.Sysctl,
 		maglev:             p.Maglev,
+		lbConfig:           p.LBConfig,
```

Comment:
More of a high-level design/org question: would it be possible to plumb this through (lb/maglev/sysctl) LocalNodeConfiguration instead? Asking because most of WriteNodeConfig will be migrated to [nodeConfig() in pkg/datapath/loader](https://github.com/cilium/cilium/blob/61a7033a3b81080a046db48122cc8a33c9e3a8bb/pkg/datapath/loader/config.go#L12) and I want to keep the porting work as straightforward as possible. 

`nodeConfig()` already receives a LNC, and I don't think making all loader entrypoints take feature config objects is a good idea. (Maybe we should also freeze additions to `HeaderfileWriter` like you've suggested here for this reason?)

Reply:
Sounds like a good idea. I’ll move it (and maybe other deps too) into LNC.

Reply:
Thanks for looking into it! Re: sysctl: makes sense, looks like it's a query interface into actual sysctl data.

Re: maglev: I think WriterParams (or, LNC actually) should embed a `maglev.Config`, not a `maglev.Maglev`. WriteNodeConfig accesses `h.maglev.Config.MaglevTableSize`, `h.maglev.SeedJhash0` and `h.maglev.SeedJhash1`, so I'd argue SeedJhash0/1 could be refactored to be a part of `maglev.Config` instead. Could that work?

Reply:
That could work yeah! Would need to split `maglev.Config` into the "user facing immutable config" and the actual `Config` that has the seeds computed, but that's straightforward. Will do another PR for this.

Reply:
https://github.com/cilium/cilium/pull/39258